hail/python/hail/context.py

```diff
@@ -687,6 +686,20 @@ def __exit__(self, type, value, traceback):
         except FileNotFoundError:
             pass
 
+    async def __aenter__(self):
+        return self.name
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ):
+        try:
+            return await self.fs.aremove(self.name)
```

Comment:
I don't think you can return in `exit`; it actually means something. [Python docs](https://docs.python.org/3/reference/datamodel.html#object.__exit__):
> If an exception is supplied, and the method wishes to suppress the exception (i.e., prevent it from being propagated), it should return a true value. Otherwise, the exception will be processed normally upon exit from this method.

Reply:
oops, thanks for the find!